tools/worker-runner/errorreport/errorreport_test.go

```diff
@@ -0,0 +1,89 @@
+package errorreport
+
+import (
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/taskcluster/taskcluster/v30/internal/workerproto"
+	ptesting "github.com/taskcluster/taskcluster/v30/internal/workerproto/testing"
+	"github.com/taskcluster/taskcluster/v30/tools/worker-runner/run"
+	"github.com/taskcluster/taskcluster/v30/tools/worker-runner/tc"
+)
+
+func TestHandleMessage(t *testing.T) {
+	description := "this is a serious error"
+	extra := map[string]interface{}{
+		"foo": "bar",
+	}
+	kind := "severe"
+	title := "test error"
+
+	t.Run("with capability", func(t *testing.T) {
+		wkr := ptesting.NewFakeWorkerWithCapabilities("error-report")
+		defer wkr.Close()
+
+		state := run.State{}
+		wkr.RunnerProtocol.Register("error-report", func(msg workerproto.Message) {
+			HandleMessage(msg, tc.FakeWorkerManagerClientFactory, &state)
+		})
+		wkr.RunnerProtocol.AddCapability("error-report")
+		wkr.RunnerProtocol.Start(false)
+		wkr.RunnerProtocol.WaitUntilInitialized()
+		wkr.WorkerProtocol.Start(true)
+		wkr.WorkerProtocol.Send(workerproto.Message{
+			Type: "error-report",
+			Properties: map[string]interface{}{
+				"description": description,
+				"extra":       extra,
+				"kind":        kind,
+				"title":       title,
+			},
+		})
+
+		// delay in between sending message and checking outcome
+		time.Sleep(5 * time.Millisecond)
```

Comment:
Instead of assuming this is long enough, it'll be less intermittent to poll in a loop until there's more than zero reports.  To help the failure case to happen quickly, it also makes sense to put a relatively short timeout on that -- something I just added for my test in https://github.com/taskcluster/taskcluster/pull/3009/commits/5ec9281d4a504e621cc73cf19fda2b1e702ac10c.